src/Microsoft.ML.AutoML/Utils/UserInputValidationUtil.cs

```diff
@@ -57,6 +57,14 @@ public static void ValidateNumberOfCVFoldsArg(uint numberOfCVFolds)
             }
         }
 
+        public static void ValidateSamplingKey(string samplingKeyColumnName, string groupIdColumnName, TaskKind task)
```

Comment:
_(appended to code to allow for threaded discussion)_

**Sugestion:** I think it should be somewhere documented on the AutoML docs that `samplingKeyColumnName` should match the `groupIdColumnName` when running ranking tasks.

Also, to sync with the ModelBuilder team to make sure that, whatever API they'll use to run Ranking experiments, they should automatically set `samplingKeyColumnName = groupIdColumnName` /cc: @LittleLittleCloud .

Reply:
Technically, the user shouldn't be providing the `SamplingKeyColumnName` in the ranking experiment, since the groupId  is used to split data. We could throw an error if the `SamplingKeyColumnName` is filled. I think a better approach is the current approach, where we only notify the user via an error if the `SamplingKeyColumnName` is different from `GroupIdColumnName`.

Reply:
What do you think about adding docs stating that for ranking AutoML will automatically use `groupIdColumnName` as a `samplingKeyColumnName` and that they can't provide `samplingKeyColumnName` that is different than `groupIdColumnName`?

Reply:
Maybe it's better to simply add a note on `ColumnInformation.GroupIdColumnName` informing users something like "If a SamplingKeyColumnName is provided, then it should be the same as GroupIdColumnName".

---
In reply to: [451216220](https://github.com/dotnet/machinelearning/pull/5263#discussion_r451216220) [](ancestors = 451216220)